### PR TITLE
ref(ts) Conver OrganizationBadge to Typescript

### PR DESCRIPTION
**Pull request by markstory**
I've removed the direct OrganizationStore subscription and instead decided to leverage the withOrganization() HOC which retains the behavior of seeing avatar updates in org settings immediately.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/shortId.tsx [hunk 1]
```diff
@@ -44,7 +45,7 @@ const StyledShortId = styled('div')`
   justify-content: flex-end;
 `;
 
-const StyledAutoSelectText = styled(AutoSelectText)<{
+const StyledAutoSelectText = styled(AutoSelectText, {shouldForwardProp: isPropValid})<{
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/shortId.tsx]: I think we need to forward if is valid prop and prop is not `avatar` (`isPropValid` is default for `shouldForwardProp`)

File: tests/js/spec/views/organizationGroupDetails/__snapshots__/groupSimilar.spec.jsx.snap [hunk 2]
```diff
@@ -1502,11 +1502,9 @@ exports[`Issues Similar View renders with mocked data 1`] = `
                                             avatar={true}
                                           >
                                             <AutoSelectText
-                                              avatar={true}
```
Comment (billyvg) [inline tests/js/spec/views/organizationGroupDetails/__snapshots__/groupSimilar.spec.jsx.snap]: o maybe not

File: src/sentry/static/sentry/app/components/idBadge/organizationBadge.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  className?: string;
+  avatarSize?: OrganizationAvatar['props']['size'];
+  // If true, will use default max-width, or specify one as a string
+  hideOverflow?: boolean | string;
+  hideAvatar?: boolean;
+};
+
+class OrganizationBadge extends React.Component<Props> {
+  static propTypes = {
+    ...BaseBadge.propTypes,
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/idBadge/organizationBadge.tsx]: Do we need a `TODO(ts)` for when we convert `BaseBadge`?

**Review by billyvg — APPROVED** [review]

File: tests/js/spec/views/organizationGroupDetails/__snapshots__/groupSimilar.spec.jsx.snap (hunk 2 above)
Comment (markstory) [reply tests/js/spec/views/organizationGroupDetails/__snapshots__/groupSimilar.spec.jsx.snap]: Yeah I was trying to remove the DOM warnings about a div getting an invalid prop now that AutoSelectText is in typescript.

File: src/sentry/static/sentry/app/components/idBadge/organizationBadge.tsx (hunk 3 above)
Comment (markstory) [reply src/sentry/static/sentry/app/components/idBadge/organizationBadge.tsx]: I'm going to try and do that whole module tree as my next block of typescript updates.

**Pull request merged**